cpp/src/arrow/python/util/datetime.h

```diff
@@ -272,9 +272,6 @@ static inline int64_t PyDate_to_days(PyDateTime_Date* pydate) {
 
 static inline int64_t PyDate_to_ms(PyDateTime_Date* pydate) {
   int64_t total_seconds = 0;
-  total_seconds += PyDateTime_DATE_GET_SECOND(pydate);
```

Comment:
an alternative approach would be to do if (pycheck_DateTime(pydate)) here, not sure if that is preferred.